x-pack/plugins/ecs_data_quality_dashboard/server/routes/results/get_results_indices_pattern.ts

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter, Logger } from '@kbn/core/server';
+
+import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import {
+  INTERNAL_API_VERSION,
+  RESULTS_INDICES_PATTERN_ROUTE_PATH,
+} from '../../../common/constants';
+import { buildResponse } from '../../lib/build_response';
+import { buildRouteValidation } from '../../schemas/common';
+import type { ResultDocument } from '../../schemas/result';
+import {
+  GetResultsIndicesPatternQuery,
+  GetResultsIndicesPatternParams,
+} from '../../schemas/result';
+import type { DataQualityDashboardRequestHandlerContext } from '../../types';
+import { API_RESULTS_INDEX_NOT_AVAILABLE } from './translations';
+import { API_DEFAULT_ERROR_MESSAGE } from '../../translations';
+import { getAuthorizedIndexNames } from '../../helpers/get_authorized_index_names';
+
+interface GetQuery {
+  indexNames: string[];
+  limit?: number;
+  from?: number;
+  outcome?: 'pass' | 'fail';
+  startDate?: string;
+  endDate?: string;
+}
+
+type ExtractArrayType<T> = T extends Array<infer U> ? U[] : never;
+type NonNullableQuery = NonNullable<SearchRequest['query']>;
+type NonNullableSort = NonNullable<SearchRequest['sort']>;
+type NonNullableQueryBool = NonNullable<NonNullableQuery['bool']>;
+type NonNullableFilter = NonNullable<NonNullableQueryBool['filter']>;
+
+export interface GetQuerySearchRequest extends SearchRequest {
+  query: NonNullableQuery & {
+    bool: NonNullableQueryBool & {
+      filter: ExtractArrayType<NonNullableFilter>;
+    };
+  };
+  sort: NonNullableSort;
+}
+
+export const getQuery = ({
+  indexNames,
+  limit,
+  from,
+  outcome,
+  startDate,
+  endDate,
+}: GetQuery): GetQuerySearchRequest => {
```

Comment:
NIT: We don't have to be very strict for the `getQuery` function return type. It is okay to change the query, as long as it is a valid `SearchRequest` object, so it can be passed to the ES client for the search request. I think it's better to use `SearchRequest` type directly, so we have the flexibility to change the implementation without having to adapt the types as well.